wafv2/rate-based-acl/main.tf

```diff
@@ -0,0 +1,107 @@
+resource "aws_wafv2_web_acl" "rate_based" {
+  name        = "rate-based"
+  description = "Rate based ACL"
+  scope       = "REGIONAL"
+
+  dynamic "default_action" {
+    for_each = var.action_default == "allow" ? ["allow"] : []
+    content {
+      allow {}
+    }
+  }
+
+  dynamic "default_action" {
+    for_each = var.action_default == "block" ? ["block"] : []
+    content {
+      allow {}
+    }
+  }
+
+
+  rule {
+    name     = "x-forwarded-for"
+    priority = 1
+
+    dynamic "action" {
+      for_each = var.action_x-forwarded-for == "allow" ? ["allow"] : []
+      content {
+        allow {}
+      }
+    }
+
+    dynamic "action" {
+      for_each = var.action_x-forwarded-for == "block" ? ["block"] : []
+      content {
+        block {}
+      }
+    }
+
+    dynamic "action" {
+      for_each = var.action_x-forwarded-for == "count" ? ["count"] : []
+      content {
+        count {}
+      }
+    }
+
+    dynamic "action" {
+      for_each = var.action_x-forwarded-for == "captcha" ? ["captcha"] : []
+      content {
+        captcha {}
+      }
+    }
+
+    statement {
+      rate_based_statement {
```

Comment:
I would rather recommend have it as another rule otherwise parametrize this would require have conditional block 
```
forwarded_ip_config
```
And I think that rule itself is already on the edge of readability.

Another point is that `aggregate_key_type = "IP"` may require another additional and conditional configuration like `geo_match_statement` which would make code even less readable. So perhaps logical switch for turning rules on and off might be better solution. What do you think?

Reply:
I had the same dilemma :) Lets discuss F2F